alf/trainers/off_policy_trainer.py

```diff
@@ -92,18 +92,27 @@ def train_iter(self, iter_num, policy_state, time_step):
 class AsyncOffPolicyTrainer(OffPolicyTrainer):
     """Perform off-policy training using AsyncOffPolicyDriver"""
 
+    def __init__(self, config):
+        super().__init__(config)
+        self._driver_started = False
+
     def init_driver(self):
+        envs = [self._env]
+        for i in range(1, self._config.num_envs):
+            envs.append(create_environment())
         driver = AsyncOffPolicyDriver(
-            env_f=create_environment,
+            envs=envs,
             algorithm=self._algorithm,
             use_rollout_state=self._config.use_rollout_state,
             unroll_length=self._unroll_length,
             debug_summaries=self._debug_summaries,
             summarize_grads_and_vars=self._summarize_grads_and_vars)
-        driver.start()
```

Comment:
why do you move driver.start() to train_iter()?

Reply:
we reset the environments and get initial timestep at the begin of training, `EnvThread` should run after that, so `env.*` will not run simultaneously